modal/aio.py

```diff
@@ -44,3 +46,14 @@
     "aio_web_endpoint",
     "aio_wsgi_app",
 ]
+
+deprecation_warning(
+    datetime.date(2023, 5, 12),
+    "The modal.aio module and Aio* classes will soon be deprecated.\n"
```

Comment:
minor but maybe put `modal.aio` and `Aio*` in backticks (otherwise they look funny in the docs online)